sigopt/hyperopt/base.py

```diff
@@ -24,9 +27,12 @@ def new_refresh():
   def parameters(self):
     return [self.trial_parameters(trial) for trial in self.trials]
 
-  def upload(self, trials=None):
+  def upload(self, trials=None, validate=False):
     new_trials = []
     trials = trials if trials is not None else self.trials
+    if validate:
+      for trial in trials:
+        self._trials.assert_valid_trial(trial)
```

Comment:
why not validate inside the other for in line 36?

Reply:
Keep validation from other logic.

Reply:
hm... that's a good point! but I don't see much benefit in this case.
you are looping over the trials to filter some of them before inserting with `_upload`, so I would argue that you already did this separation. Might as well call the modular `assert_valid_trial` during that loop.

but it's up to you, the overhead is negligible

Reply:
Validation is not necessary if the `trials` is refreshed by `fmin`, as `fmin` will validate trials, it may be only necessary when user uses `upload_trials`. So I do not think we need to always validate.